u8timeseries/utils/timeseries_generation.py

```diff
@@ -0,0 +1,137 @@
+    :return: A constant TimeSeries with value 'value'.
+    """
+
+    times = pd.date_range(periods=length, freq=offset, start=start_date)
+    values = np.full(length, value)
+
+    return TimeSeries.from_times_and_values(times, values)
+
+
+def linear_timeseries(start_value: float = 0, value_delta: float = 1, length: int = 10, offset: str = 'D',
+                      start_date: pd.Timestamp = pd.Timestamp('2000-01-01')) -> 'TimeSeries':
+    """
+    Creates a TimeSeries with a starting value of 'start_value' that increases by 'value_delta' at each step.
+    The last entry of the time series will be equal to 'start_value' + ('length' - 1) * 'value_delta'.
+
+    :param start_value: The value of the first entry in the TimeSeries.
+    :param value_delta: The difference in value between to adjacent entries in the TimeSeries.
+    :param length: The length of the returned TimeSeries.
+    :param offset: The time differene between two adjacent entries in the returned TimeSeries. A DateOffset alias is expected.
+    :param start_date: The time index of the first entry in the returned TimeSeries.
+    :return: A linear TimeSeries with gradient 'value_delta'.
+    """
+
+    times = pd.date_range(periods=length, freq=offset, start=start_date)
+    values = np.linspace(start_value, start_value + (length - 1) * value_delta, length)
+
+    return TimeSeries.from_times_and_values(times, values)
+
+
+def periodic_timeseries(frequency: float = 0.1, amplitude: float = 1, phase: float = 0, y_offset: float = 0,
+                        length: int = 10, offset: str = 'D',
+                        start_date: pd.Timestamp = pd.Timestamp('2000-01-01')) -> 'TimeSeries':
+    """
+    Creates a TimeSeries with a sinusoidal value progression with a given frequency, amplitude, phase and y offset.
+
+    :param frequency: The number of periods that take place within one time unit given in 'offset'.
+    :param amplitude: The maximum  difference between any value of the returned TimeSeries and 'y_offset'.
+    :param phase: The relative position within one period of the first value of the returned TimeSeries (in radians).
+    :param y_offset: The shift of the sine function along the y axis.
+    :param length: The length of the returned TimeSeries.
+    :param offset: The time differene between two adjacent entries in the returned TimeSeries. A DateOffset alias is expected.
+    :param start_date: The time index of the first entry in the returned TimeSeries.
+    :return: A sinusoidal TimeSeries parametrized as indicated above.
+    """
+
+    times = pd.date_range(periods=length, freq=offset, start=start_date)
+    values = np.array(range(length), dtype=float)
+    f = np.vectorize(lambda x: amplitude * math.sin(2 * math.pi * frequency * x + phase) + y_offset)
+    values = f(values)
+
+    return TimeSeries.from_times_and_values(times, values)
+
+
+def white_noise_timeseries(length: int = 10, offset: str = 'D', mean: float = 0, std: float = 1,
```

Comment:
How about calling this one `gaussian_timeseries` ? As when `mean` is not 0 this wouldn't (technically) be white noise.

Reply:
You could even optionally support the case where `mean` is a vector and `std` is a matrix, and the resulting distribution is a multivariate Gaussian. If `mean` and `std` are scalars you could revert to of i.i.d samples as you have now.